### PR TITLE
update: errbot-ified urlmatcher plugin

### PR DESCRIPTION
**Pull request by mrshu**
- Made the urlmatcher plugin work with errbot.

Signed-off-by: mr.Shu mr@shu.io

---

Note that this plugin is the [`errbot`](http://errbot.net)'s version of the brutal's [url matcher plugin](https://github.com/mrshu/brutal-plugins/blob/master/url_matcher.py).

### CODE REVIEW

**Comment by mrshu** [comment]
to @pepol @Adman @JakubNvk 

cc @gbin @tWido

File: err-urlmatcher/urlmatcher.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+# see https://github.com/mrshu/brutal-plugins/issues/38
+URL_REGEX = '.*((:?https?|ftp)://[^\s/$.?#].[^\s\x0f]*).*'
+TAG_RE = re.compile(r'<[^>]+>')
+WHITESPACE_RE = re.compile(r'\s\s+')
+
+
+class UrlMatcher(BotPlugin):
+    @re_botcmd(pattern=URL_REGEX, prefixed=False)
+    def url_matcher(self, msg, match):
+        url = match.group(0)
```
Comment (mrshu) [inline err-urlmatcher/urlmatcher.py]: I particularly do not like this line and would love to get the `url` as a parameter to this function (possibly the very last one, after `match`).

File: err-urlmatcher/urlmatcher.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+            return
+
+        html = requests.get(url).text
+        readable_article = Document(html).summary()
+        readable_article = TAG_RE.sub('', readable_article)
+        readable_article = WHITESPACE_RE.sub(' ', readable_article)
+        readable_article = readable_article.replace('\n', ' ')
+        readable_article = readable_article.replace('&#13;', '')
+
+        if len(readable_article) > 75:
```
Comment (mrshu) [inline err-urlmatcher/urlmatcher.py]: The `75` here could be configurable.

File: err-urlmatcher/urlmatcher.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+
+class UrlMatcher(BotPlugin):
+    @re_botcmd(pattern=URL_REGEX, prefixed=False)
+    def url_matcher(self, msg, match):
+        url = match.group(0)
+        r = requests.head(url)
+        # files that are too big cause trouble. Let's just ignore them.
+        if 'content-length' in r.headers and \
+           int(r.headers['content-length']) > 5e6:
```
Comment (mrshu) [inline err-urlmatcher/urlmatcher.py]: This `5e6` here could also be configurable, with `5e6` being the default.

**Comment by gbin** [comment]
cool !

**Pull request merged**